src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/QueryStringSegment.cs

```diff
@@ -7,7 +7,14 @@ public class QueryStringSegment : PatternSegment
     {
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return context.HttpContext.Request.QueryString.ToString();
+            var queryString = context.HttpContext.Request.QueryString.ToString();
```

Comment:
There's actually no `string.IsEmpty` method in C#. It would have to be a string compare to the `string.Empty` field. I think I'll just leave null and empty check 😄